src/StudyUtils.in.jsm

```diff
@@ -76,52 +143,107 @@ function merge(source) {
         Object.keys(object);
     return [...names, ...symbols];
   }
+  /*
+  * descriptor: an object whose own enumerable properties constitute descriptors
+  * for the properties from arguments[1]+ to be defined or modified in
+  * arguments[0]
+  */
   const descriptor = {};
-
-  // `Boolean` converts the first parameter to a boolean value. Any object is
-  // converted to `true` where `null` and `undefined` becames `false`. Therefore
-  // the `filter` method will keep only objects that are defined and not null.
-  Array.slice(arguments, 1).filter(Boolean).forEach(function onEach(properties) {
-    getOwnPropertyIdentifiers(properties).forEach(function(name) {
+  /*
+  * `Boolean` converts the first parameter to a boolean value. Any object is
+  * converted to `true` where `null` and `undefined` becames `false`. Therefore
+  * the `filter` method will keep only objects that are defined and not null.
+  */
+  Array.slice(arguments, 1).filter(Boolean).forEach((properties) => {
+    getOwnPropertyIdentifiers(properties).forEach((name) => {
       descriptor[name] = Object.getOwnPropertyDescriptor(properties, name);
     });
   });
   return Object.defineProperties(source, descriptor);
 }
 
+/**
+* Appends a query string to a url.
+* @param {string} url - a base url to append; must be static (data) or external
+* @param {Object} args - query arguments, one or more object literal used to
+* build a query string
+* @returns {string} - an absolute url appended with a query string
+*/
 function mergeQueryArgs(url, ...args) {
-  /* currently left to right*/
+  // currently left to right
   // TODO, glind, decide order of merge here
   // TODO, use Object.assign, or ES7 spread
   const U = new URL(url);
+  // get the query string already attached to url, if it exists
   let q = U.search || "?";
+  // create an interface to interact with the query string
   q = new URLSearchParams(q);
-
   const merged = merge({}, ...args);
-
-  // get user info.
+  // Set each search parameter in "merged" to its value in the query string,
+  // building up the query string one search parameter at a time.
   Object.keys(merged).forEach((k) => {
+    // k, the search parameter (ex: fxVersion)
+    // q.get(k), returns the value of k, in query string, q (ex: 57.0.1a)
     log.debug(q.get(k), k, merged[k]);
     q.set(k, merged[k]);
   });
-
+  // append our new query string to the URL object made with "url"
   U.search = q.toString();
+  // return the full url, with the appended query string
   return U.toString();
 }
 
 // sampling utils
+/**
+* @async
+* Converts a string into its sha256 hexadecimal representation.
+* Note: This is ultimately used to make a hash of the user's telemetry clientID
+* and the study name.
+* @param {string} message - The message to convert.
+* @returns {string} hashHex - a hexadecimal, 256-bit hash
```

Comment:
For `@returns`, I dont think you need the variable name.
If that's right, we have a few instances of this:

```sh
$ git grep '@returns {' | grep -v '} - ' | grep -v '{void}'

src/StudyUtils.in.jsm:* @returns {string} hashHex - a hexadecimal, 256-bit hash
src/StudyUtils.in.jsm:  * @returns {string} id - the telemetry client ID
src/StudyUtils.in.jsm:  * @returns {Object} variation - the study variation for this user
src/StudyUtils.in.jsm:  * @returns {Object} queryArgs - the query arguments for the study
src/StudyUtils.in.jsm:* @returns {Object} L - the Logger instance, see gre/modules/Log.jsm
```

Reply:
It is optional; I've removed it for more clarity.